### ISSUE TITLE
Check the bts airdrop replacement after snapshot

### ISSUE THREAD

**Issue by HackFisher** [bitsuperlab/bitshares_play#45]
https://github.com/BitShares/bitshares/commit/28dab197914f07a0c4e7657d4ab1c8a6db985c06#diff-259729ccb429a4b4d5d877d9660dae1f

https://github.com/BitShares/bitshares/commits/efee28999b6d7a02e2232431b7c210063d13a768/libraries/blockchain/genesis_bts.json

https://github.com/BitShares/bitshares/commit/efee28999b6d7a02e2232431b7c210063d13a768#diff-259729ccb429a4b4d5d877d9660dae1f

1. get the relate bts address of the pts/btc address in replacement
2. check it in libraries/blockchain/pts_ags_snapshot/bts_20141208.json the bts address should be the one in finalized (after replacement)

**Comment by clar** [comment]
@HackFisher , i'm checking this issue. we have a tool bts_key_info which can get 'native_addr' 'pts_address' from 'public_key', but when I regenerate 'native_addr' from a given 'pts_address', the native_addr is changed, do you know the reason?

**Comment by clar** [comment]
fc::mutable_variant_object obj;

            obj["public_key"] = key;
            obj["native_address"] = bts::blockchain::address(key.operator fc::ecc::public_key_data());
            obj["pts_address"] = bts::blockchain::pts_address(key.operator fc::ecc::public_key());
            
            fc::mutable_variant_object obj2;
            string raw_pts_addr = obj["pts_address"].as_string();
            pts_address pts_addr = pts_address(raw_pts_addr);
            obj2["pts_address"] = pts_addr;
            obj2["native_address"] = bts::blockchain::address(pts_addr);

**Comment by clar** [comment]
there is no way to get related bts address from a pts/btc address, basically bts address is hash of a public key and pts address is double hash of pub key, the crypto hash is one way.

So only the person who owns the private key of that address, can the public key and bts address, by wallet_import_private_key api, and then check if the bts address is in our genesis file.

I checked that the bts genesis file we used to do snapshot contains the substitution, so this should be not a problem.
